pkg/controller/sync/controller.go

```diff
@@ -373,3 +364,123 @@ func (s *FederationSyncController) syncToClusters(fedResource FederatedResource,
 
 	return status
 }
+
+func (s *FederationSyncController) ensureDeletion(fedResource FederatedResource) error {
+	fedResource.DeleteVersions()
+
+	key := fedResource.FederatedName().String()
+	kind := fedResource.FederatedKind()
+
+	glog.V(2).Infof("Ensuring deletion of %s %q", kind, key)
+
+	obj := fedResource.Object()
+
+	finalizers := sets.NewString(obj.GetFinalizers()...)
+	if !finalizers.Has(FinalizerSyncController) {
+		glog.V(2).Infof("%s %q does not have the %q finalizer. Nothing to do.", kind, key, FinalizerSyncController)
+		return nil
+	}
+
+	annotations := obj.GetAnnotations()
+	orphanResources := annotations != nil && annotations[OrphanManagedResources] == "true"
+	if orphanResources {
+		glog.V(2).Infof("Found %q annotation on %s %q. Removing the finalizer.", OrphanManagedResources, kind, key)
+		err := s.removeFinalizer(fedResource)
+		if err != nil {
+			return errors.Wrapf(err, "failed to remove finalizer %q", OrphanManagedResources)
+		}
+		glog.V(2).Infof("Initiating the removal of the label %q from resources previously managed by %s %q.", util.ManagedByFederationLabelKey, kind, key)
+		return nil
+		// TODO(marun) Implement removal of labels
+	}
+
+	glog.V(2).Infof("Deleting resources managed by %s %q from member clusters.", kind, key)
+
+	// First retrieve a list of unready clusters to maximize the
+	// changes of detecting when it's not possible to guarantee
+	// removal from member clusters.
```

Comment:
chances?

either ways it isn't clear to me what this comment means.

Reply:
I've switched to testing for unready clusters in the loop, so this comment is no longer relevant.